src/sqlite_db.cpp

```diff
@@ -207,13 +207,15 @@ SQLite::result_type SQLite::Exec(const std::string& query) const
     SQLite::result_type res;
     MIOPEN_LOG_T(std::this_thread::get_id() << ":" << query);
     {
+        // NOLINTBEGIN(clang-analyzer-core.CallAndMessage)
```

Comment:
@atamazov could you try to fix this? It is complaining about Line-213

Reply:
@junliume I guess you mean line 213 in the _original_ text, i.e. passing an address to `find_callback`. I highly recommend moving NOLINTBEGIN/END just aroung the guilty line.

This looks like a false positive because the context is very complex: it's like passing an address of a static function to another function, which is called from lambda passed to some 3rd function and invoked by it.

We can try to simplify things for the analyzer. E.g. introduce some intermediate static variable in the `SQLite` class, e.g. `find_callback_address`, initialize it with the address of `find_callback` and use it in the lambda instead of `find_callback`. You may also need to capture it by-value.

I can help more when I am back.

Reply:
Hi @atamazov thanks! It is actually complaining about this line:
```
query.c_str()
```
which is strange and also might be false alarm.

Reply:
@junliume then it is likely that analyzer is unable to track the actual lifetime of `query`. To check this hypothesis, just pass it to lambda by-value.

This is false alarm, UNLESS `sqlite3_exec` is doing something unusual, like putting the db access request into some queue, so actual execution may be postponed.